### ISSUE TITLE
Variable substitution for python.analysis.stubPath

### ISSUE THREAD

**Issue by dianlujitao** [fannheyward/coc-pyright#595]
**What's the output of `:CocCommand pyright.version`**
```
[coc.nvim] coc-pyright 1.1.161 with Pyright 1.1.161
```

**What's the output of `:CocCommand workspace.showOutput Pyright`**
```
Workspace: /home/dianlujitao/foo/bar
Using python from /usr/bin/python

[Info  - 下午3:45:01] Pyright language server 1.1.161 starting
[Info  - 下午3:45:01] Server root directory: /home/dianlujitao/.config/coc/extensions/node_modules/coc-pyright/node_modules/pyright/dist/
[Info  - 下午3:45:01] No configuration file found.
[Info  - 下午3:45:01] No pyproject.toml file found.
[Info  - 下午3:45:01] Setting pythonPath for service "bar": "/usr/bin/python"
-----------------ATTENTION
Case 1: python.analysis.stubPath is not set and defaults to typings dir of the current workspace, expected
[Warn  - 下午3:45:01] stubPath /home/dianlujitao/foo/bar/typings is not a valid directory.

Case 2:   "python.analysis.stubPath": "~/.vim/python-type-stubs", parsed as a relative path, needs fix
[Warn  - 下午3:47:24] stubPath /home/dianlujitao/foo/bar/~/.vim/python-type-stubs is not a valid directory.

Case 3:   "python.analysis.stubPath": "$HOME/.vim/python-type-stubs", parsed as a relative path, needs fix
[Warn  - 下午3:49:29] stubPath /home/dianlujitao/foo/bar/$HOME/.vim/python-type-stubs is not a valid directory.
-----------------
[Info  - 下午3:45:01] Assuming Python version 3.9
[Info  - 下午3:45:01] Assuming Python platform Linux
Exception received when installing recursive file system watcher: TypeError [ERR_FEATURE_UNAVAILABLE_ON_PLATFORM]: The feature watch recursively is unavailable on the current platform, which is being used to run Node.js
Exception received when installing recursive file system watcher: TypeError [ERR_FEATURE_UNAVAILABLE_ON_PLATFORM]: The feature watch recursively is unavailable on the current platform, which is being used to run Node.js
Exception received when installing recursive file system watcher: TypeError [ERR_FEATURE_UNAVAILABLE_ON_PLATFORM]: The feature watch recursively is unavailable on the current platform, which is being used to run Node.js
[Info  - 下午3:45:01] Searching for source files
[Info  - 下午3:45:01] Found 1 source file
```

`"python.analysis.stubPath": "/home/dianlujitao/.vim/python-type-stubs"` works but not portable enough, obviously.

I'm unsure if the issue occurs with `python.analysis.typeshedPaths` and other path variables.

The point is to use shared stub files for all python projects.

**Comment by fannheyward** [comment]
@dianlujitao released in v1.1.163, please give it a try.

**Comment by dianlujitao** [comment]
Thanks for the fix, works as expected now.
